runtime/near-vm-runner/src/imports.rs

```diff
@@ -455,13 +455,14 @@ pub(crate) mod wasmtime {
         };
     }
 
-    pub(crate) fn link(
+    pub(crate) unsafe fn link(
         linker: &mut wasmtime::Linker,
         memory: wasmtime::Memory,
-        raw_logic: *mut c_void,
+        raw_logic_plus_memory: *mut c_void,
```

Comment:
Preexisting, but I would argue that the context should be passed in as a proper type here, and that the thread local should own the entire `VMLogicPlusMemory` rather than doing this raw pointer contraption.

If possible.

Reply:
I did it this way just to get the refactor out of the way, with https://github.com/near/nearcore/pull/6127 the whole unsafe thing gets removed away, as well as wasmtime's usage of VMLogicPlusMemory :)